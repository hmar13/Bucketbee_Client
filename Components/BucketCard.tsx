import React from 'react';
import theme from '../styles/theme.style';
import {
  View,
  Text,
  StyleSheet,
  TouchableHighlight
} from 'react-native';
import moment from 'moment';
import Bucket from '../interfaces/Bucket';

interface BucketCardProps {
  bucket: Bucket;
  onPress(): void;
}

const BucketCard: React.FC<BucketCardProps> = ({ bucket, onPress }) => {
  return (
    <TouchableHighlight
      style={{ borderRadius: 25, marginVertical: 18 }}
      onPress={onPress}
    >
      <View style={styles.outercard}>
        <View style={styles.card}>
          <Text style={styles.text} >{bucket.title}</Text>
          <Text style={styles.subtext}>
            {moment(bucket.date_created).format('MMM Do, YYYY')}
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  outercard: {
    backgroundColor: theme.PRIMARY_COLOR_XLITE,
    borderRadius: 25,
    height: 202,
  },
  card: {
    height: 200,
    width: '100%',
    padding: 20,
    borderRadius: 25,
    marginVertical: 5,
    backgroundColor: '#fffef8',
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.3,
    shadowRadius: 10,
  },
  text: {
    fontFamily: 'Raleway_600SemiBold',
    fontSize: 30,
    color: theme.PRIMARY_COLOR,
    paddingVertical: 5,
  },
  subtext: {
    fontFamily: 'Poppins_300Light',
    color: theme.PRIMARY_COLOR,
  },
});

export default BucketCard;
