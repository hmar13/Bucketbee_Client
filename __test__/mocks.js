import theme from '../styles/theme.style';

const bucket = {
  __typename: 'Bucket',
  categories: [{
    __typename: 'Category',
    id: '604ca4e87f7d89153b39d99f',
    label: "Hello",
    places: [],
  }],
  date_created: '2021-03-13T11:41:28.273Z',
  id: "604ca4e87f7d89153b39d9a0",
  notes: "Hello",
  title: "Hello",
}
const userA = {
  _typename: 'User',
  birthday: null,
  createdAt: '2021-03-12T12:05:53.895Z',
  email: 'A',
  emojis: ':) ;) :O',
  firstName: 'A',
  friends: [{
    __typename: "User",
      birthday: null,
      createdAt: "2021-03-11T17:25:27.762Z",
      email: 'B',
      firstName: 'B',
      id: '604a52875a22a656d698c988',
      lastName: null,
      username: 'B',
  }],
  id: '604b592130793c1bff327555',
  lastName: null,
  location: 'London',
  profile_pic: null,
  updatedAt: '2021-03-15T09:02:58.715Z',
  username: 'A',
  vibe: 'Beach'
}
const userB = {
  _typename: 'User',
  birthday: null,
  createdAt: '2021-03-12T12:10:53.895Z',
  email: 'B',
  emojis: ':) :O :P',
  firstName: 'B',
  friends: [{
    __typename: "User",
      birthday: null,
      createdAt: "2021-03-12T12:05:53.895Z",
      email: 'A',
      firstName: 'A',
      id: '604b592130793c1bff327555',
      lastName: null,
      username: 'A',
  }],
  id: '604a52875a22a656d698c988',
  lastName: null,
  location: 'London',
  profile_pic: null,
  updatedAt: '2021-03-15T09:02:58.715Z',
  username: 'B',
  vibe: 'Beach'
}

const place = {
  __typename: 'Place',
  description: 'locality, political',
  formatted_address: 'London, UK',
  id: '',
  imgArr: [
    'https://lh3.googleusercontent.com/p/AF1QipOmH4cmfDXKSZsY0ECNfEVo99-sJ9JiwW0hOE5y=s1600-h619',
    'https://lh3.googleusercontent.com/p/AF1QipPlfRNdjQbAgVXJ6CO8xm6ArPNtIyIK-SqHx4V1=s1600-h911',
    'https://lh3.googleusercontent.com/p/AF1QipMGBhjSmSICwYc2uMiFQ0l-bZm-Yx69est4gBvO=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipMx47l1JKY-Nj9p6u5yFIj-TI7ubxM6m1_rdwSH=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipO1gRyYKC04M2jwJh6mSeuDvzhXqks1BJZW7_Yw=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipO5rYsQ8pjrN9Tmspu1uSKtLbyy4wZvO_bMyzH1=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipNB7-V6PQypu5RSJnazhue3BNaHtiBA5bqqYI5Z=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipNk5QU10zjui5OrIQ2WuMLuWChDJDmu0sGz-Zi7=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipMRx0je916q6jCTwBi3lU5AcWNHIa4dHa91eDZq=s1600-h1200',
    'https://lh3.googleusercontent.com/p/AF1QipNKpEuPRT3zlCBdgjvK5zq1-LB1TlcOCNMAyhCl=s1600-h1200',
  ],
  international_phone_number: undefined,
  latitude: 51.5073509,
  longitude: -0.1277583,
  name: 'London',
  notes: '',
  open_now: undefined,
  rating: undefined,
  review: undefined,
  url: 'https://maps.google.com/?q=London,+UK&ftid=0x47d8a00baf21de75:0x52963a5addd52a99',
  user_ratings_total: undefined,
  weekday_text: undefined,
}

const category = {
  __typename: 'Category',
  id: '604b597030793c1bff327556',
  label: 'Music',
  places: [place],
}

const message = {
  __typename: 'Message',
  author: '604b592130793c1bff327555',
  content: 'TEST TEST TEST',
  createdAt: '2021-03-15T09:05:56.583Z',
  description: 'message',
  id: '604f23740ea199065c8a602f',
  photo: null,
  timeslots: []
}
const chat = {
  __typename: 'Chat',
  admin: null,
  createdAt: '2021-03-15T09:05:50.079Z',
  id: '604f236e0ea199065c8a602e',
  members: [userA, userB],
  messages: [message],
  name: 'B',
  updatedAt: '021-03-15T16:05:48.012Z',
}

const radioButtons = [
  {
    id: '1', // acts as primary key, should be unique and non-empty string
    label: 'Beach',
    value: 'option1',
    labelStyle: {
      color: theme.PRIMARY_COLOR_XLITE,
      fontSize: 20,
      fontFamily: 'Lato_300Light',
    },
    color: theme.PRIMARY_COLOR_XLITE,
    size: 20,
    containerStyle: {
      alignSelf: 'flex-start',
    },
  },
  {
    id: '2',
    label: 'Mountains',
    value: 'option2',
    color: theme.PRIMARY_COLOR_XLITE,
    labelStyle: {
      color: theme.PRIMARY_COLOR_XLITE,
      fontSize: 20,
      fontFamily: 'Lato_300Light',
    },
    size: 20,
  },
  {
    id: '3',
    label: 'City',
    value: 'option3',
    color: theme.PRIMARY_COLOR_XLITE,
    size: 20,
    labelStyle: {
      color: theme.PRIMARY_COLOR_XLITE,
      fontSize: 20,
      fontFamily: 'Lato_300Light',
    },
    selected: true,
  },
];

module.exports = { bucket, userA, userB, place, category, message, chat, radioButtons }