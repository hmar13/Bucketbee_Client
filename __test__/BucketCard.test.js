import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import BucketCard from '../Components/BucketCard';
import { bucket } from './mocks';
import { shallow } from 'enzyme';
import moment from 'moment'

describe('<BucketCard>', () => {
  const onPress = jest.fn()
  const wrapper = shallow(<BucketCard onPress={onPress} bucket={bucket} />);

  test('Renders correctly', () => {
    const tree = renderer.create(<BucketCard onPress={onPress} bucket={bucket} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the correct information', () => {
    expect(wrapper.find('Text').at(0).props().children).toEqual(bucket.title)
    expect(wrapper.find('Text').at(1).props().children).toEqual(moment(bucket.date_created).format('MMM Do, YYYY'))
  })
  it('should call onPress', () => {
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalled();
  })

});
