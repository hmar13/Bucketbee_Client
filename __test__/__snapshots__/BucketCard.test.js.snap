// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BucketCard> Renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 25,
      "marginVertical": 18,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#9affdc",
        "borderRadius": 25,
        "height": 202,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fffef8",
          "borderRadius": 25,
          "height": 200,
          "justifyContent": "center",
          "marginVertical": 5,
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0d525f",
            "fontFamily": "Raleway_600SemiBold",
            "fontSize": 30,
            "paddingVertical": 5,
          }
        }
      >
        Hello
      </Text>
      <Text
        style={
          Object {
            "color": "#0d525f",
            "fontFamily": "Poppins_300Light",
          }
        }
      >
        Mar 13th, 2021
      </Text>
    </View>
  </View>
</View>
`;
